python/BUILDING.md

```diff
@@ -1,220 +1,151 @@
-configuration:
+To build a debug version for use with `python_d`, set the `Configuration` property to `Debug`:
 
 ```shell
-msbuild msbuild\ice.proj /p:Configuration=Release /p:Platform=Win32
+MSBuild msbuild\ice.proj /p:Configuration=Debug
 ```
 
-When using the MSBuild Platform property, the build platform doesn't depend
-on the command prompt's default platform.
-
-The build will use the default location for Python defined in
-`python\msbuild\ice.props`. You can override it by setting the `PythonHome`
-MSBuild property. For example, the following command will use the Python
-installation from `C:\Python312-AMD64` instead of the default location:
+To change the target platform, use the `Platform` property. For example, to build for `Win32` in debug mode:
 
 ```shell
-msbuild msbuild\ice.proj /p:Configuration=Release /p:Platform=x64 /p:PythonHome=C:\Python312-AMD64
+MSBuild msbuild\ice.proj /p:Platform=Win32 /p:Configuration=Debug
 ```
 
-## Building on Linux or macOS
-
-Ice for Python supports Python versions 3.12 and greater. Note however that
-your Python installation must have been built with a C++ compiler that is
-compatible with the compiler used to build Ice for C++.
+> ⚠️ *The `Platform`, `Configuration`, and `PythonHome` settings must match your C++ build and Python version.*
 
-The build of Ice for Python requires to first build Ice for C++ in the `cpp`
-subdirectory.
+### Running the tests
 
-From the top-level source directory, edit `config/Make.rules` to establish your
-build configuration. The comments in the file provide more information.
-
-Change to the Ice for Python source subdirectory:
+To run the tests, open a command prompt and change to the `python` subdirectory. Then run:
 
 ```shell
-cd python
+python allTests.py --all
 ```
 
-Execute `python -V` to verify that the correct Python interpreter is in your
-executable search path.
+### Generating the API reference
 
-Run `make` to build the extension.
+To build the API reference documentation, run the following commands from the `python/docs` subdirectory:
 
-Upon successful completion, run `make install`. You may need additional user
-permissions to install in the directory specified by `config/Make.rules`.
+1. Create and activate a Python virtual environment:
 
-## Configuring your Environment for Python
+   ```shell
+   python -m venv venv
+   source venv/bin/activate  # On macOS/Linux
+   venv\Scripts\activate     # On Windows
+   ```
 
-Modify your environment to allow Python to find the Ice extension for Python.
-The python interpreter must be able to locate the IcePy extension as well as
-the Python source files in the `python` subdirectory. This is normally
-accomplished by setting the `PYTHONPATH` environment variable to contain the
-necessary subdirectory.
+2. Install the documentation dependencies:
 
-For example on Windows, with Ice for Python installed in `C:\Ice-3.8.0`:
+   ```shell
+   pip install -r requirements.txt
```

Comment:
I believe it's still a good idea to do `python(3?) -m pip`

Reply:
Does it change anything when we are in a virtual environment?